src/App.tsx

```diff
@@ -15,36 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetical = 'alphabetical',
+  length = 'length',
+}
+
+function getPrepareGoods(
+  goods: string[],
+  field: string,
+  reversed: boolean,
+): string[] {
+  let prepareGoods = [...goods];
+
+  if (field === SortType.alphabetical) {
+    prepareGoods.sort((a: string, b: string) => a.localeCompare(b));
+  } else if (field === SortType.length) {
+    prepareGoods.sort((a: string, b: string) => a.length - b.length);
+  }
+
+  if (reversed) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReversed] = useState(false);
+
+  const updatedGoods = getPrepareGoods(goodsFromServer, sortField, isReverse);
+
+  const handleclick = (field: string) => {
+    if (field === sortField) {
+      setIsReversed(!isReverse);
+      setSortField(field);
+    }
```

Comment:
- When I click on any sort type button it toggles `reverse`, and it doesn't make sense 🤷. The Sort button should only set the corresponding sort type, reverse button toggle the `isReversed` state.
- The name is not good too, what click does it handle?
- When you need to change state depending on the previous one use functional `setState`. `setIsReversed(prev => !prev)`